### PR TITLE
feat port: Order.Antisymmetrization

### PR DESCRIPTION
**Pull request by ChrisHughes24**
Mathlib SHA f1a2caaf51ef593799107fe9a8d5e411599f3996

There were a bunch of changes from `toAntisymmetrization (. ≤ .)` to `@toAntisymmetrization α (· ≤ ·) _` and also the coercion to fun for `OrderIso.dualAntisymmetrization α` did not trigger without making `α` explicit, previously there was an underscore in place of `α`

### CODE REVIEW

File: Mathlib/Order/Antisymmetrization.lean [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+  ⟨Quotient.map' f <| lift_fun_antisymm_rel f, fun a b => Quotient.inductionOn₂' a b <| f.mono⟩
+#align order_hom.antisymmetrization OrderHom.antisymmetrization
+
+@[simp]
+theorem OrderHom.coe_antisymmetrization (f : α →o β) :
+    f.antisymmetrization = Quotient.map' f (lift_fun_antisymm_rel f) :=
+  rfl
+#align order_hom.coe_antisymmetrization OrderHom.coe_antisymmetrization
+
+/- Porting notes: Removed @[simp] attribute -/
```
Comment (semorrison) [inline Mathlib/Order/Antisymmetrization.lean:241]: Probably best if you say why in the porting note.

File: Mathlib/Order/Antisymmetrization.lean [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+@[simps]
+noncomputable def OrderEmbedding.ofAntisymmetrization :
+    Antisymmetrization α (· ≤ ·) ↪o
+      α where
+  toFun := _root_.ofAntisymmetrization (. ≤ .)
+  inj' _ _ := Quotient.out_inj.1
+  map_rel_iff' := ofAntisymmetrization_le_ofAntisymmetrization_iff
+#align order_embedding.of_antisymmetrization OrderEmbedding.ofAntisymmetrization
+
+/-- `antisymmetrization` and `order_dual` commute. -/
```
Comment (semorrison) [inline Mathlib/Order/Antisymmetrization.lean:265]:
```suggestion
/-- `Antisymmetrization` and `OrderDual` commute. -/
```

File: Mathlib/Order/Antisymmetrization.lean [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+variable (α) (r : α → α → Prop) [IsPreorder α r]
+
+/-- The antisymmetrization relation as an equivalence relation. -/
+@[simps]
+def AntisymmRel.setoid : Setoid α :=
+  ⟨AntisymmRel r, antisymmRel_refl _, AntisymmRel.symm, AntisymmRel.trans⟩
+#align antisymm_rel.setoid AntisymmRel.setoid
+
+/-- The partial order derived from a preorder by making pairwise comparable elements equal. This is
+the quotient by `λ a b, a ≤ b ∧ b ≤ a`. -/
```
Comment (semorrison) [inline Mathlib/Order/Antisymmetrization.lean:87]:
```suggestion
/-- The partial order derived from a preorder by making pairwise comparable elements equal. This is
the quotient by `fun a b => a ≤ b ∧ b ≤ a`. -/
```

**Comment by semorrison** [comment]
bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by ChrisHughes24** [comment]
This seems to have been merged without the PR being closed. Closing it now

**Pull request closed without merging**